File: rest-common/src/main/java/org/chronopolis/rest/api/IngestAPIProperties.java
package org.chronopolis.rest.api;

import org.springframework.boot.context.properties.ConfigurationProperties;

import java.util.List;

/**
 * Properties for connecting to an ingest api
 *
 * @author shake
 */
@ConfigurationProperties(prefix = "ingest.api")
public class IngestAPIProperties {

    /**
     * The endpoint of the Ingest Server
     */
    private String endpoint = "http://localhost:8080/ingest/";

    /**
     * The username to connect to the Ingest Server as
     */
    private String username = "ingest-user";

    /**
     * The password to use
     */
    private String password = "change-me";

    /**
     * List of nodes to replicate to
     */
    private List<String> replicateTo;

    public String getEndpoint() {
        return endpoint;
    }

    public IngestAPIProperties setEndpoint(String endpoint) {
        this.endpoint = endpoint;
        return this;
    }

    public String getUsername() {
        return username;
    }

    public IngestAPIProperties setUsername(String username) {
        this.username = username;
        return this;
    }

    public String getPassword() {
        return password;
    }

    public IngestAPIProperties setPassword(String password) {
        this.password = password;
        return this;
    }

    public List<String> getReplicateTo() {
        return replicateTo;
    }

    public IngestAPIProperties setReplicateTo(List<String> replicateTo) {
        this.replicateTo = replicateTo;
        return this;
    }

}

File: rest-models/src/main/kotlin/org/chronopolis/rest/api/IngestApiProperties.kt
package org.chronopolis.rest.api

import org.springframework.boot.context.properties.ConfigurationProperties

/**
 * Properties for connecting to a Chronopolis Ingest Server
 *
 * @property endpoint The endpoint of the Ingest Server
 * @property username The username to connect to the Ingest Server as
 * @property password The password to use
 *
 * @author shake
 */
@ConfigurationProperties(prefix = "ingest.api")
data class IngestApiProperties(var endpoint: String = "http://localhost:8080/",
                               var username: String = "ingest-user",
                               var password: String = "change-me")
